### DISCUSSION TITLE
Problem using fine resolution DEM

### DISCUSSION THREAD

**Discussion by maawoo** [davidfrantz/force#68] [Q&A]
Hi there!

I'm running into problems (or at least **very** long processing times) with a 5m DEM, which was derived from Lidar data. 
To rule out other issues, I processed the same two Landsat 8 scenes with the same processing settings using a 30m/1 arc second SRTM (uncompressed; Int16; EPSG:4326), a 10m Lidar DEM (uncompressed; Float32; Glance7) and the 5m Lidar DEM (uncompressed; Float32; Glance7). 
The last two are basically the same except for the spatial resolution, as both were derived using the same GDAL command and just changing the values for target resolution (`-tr` flag).  

30m SRTM: ~1min 54sec per scene
10m Lidar: ~1min 43sec per scene 
5m Lidar: I stopped the process after 30min... 🤷‍♂️

So the slightly faster processing time for the 10m DEM was a positive surprise, but over 30min using the 5m DEM is not normal, right? Any ideas?

While your docs state the following: 

> The DEM should match the resolution of the Level 1 image data as closely as possible. **If possible, it is advised to use a finer resolution.**

...I'm not sure if the benefit would be worth it anyway? At least not for Landsat 8, but I'm also going to process Sentinel-2 scenes of course.

**Comment by davidfrantz** [comment]
Hi Marco, 

I guess that the 10m DEM is more than sufficient in this case.

However, I am also surprised by this huge increase in processing time.. 

- Do you use the same target resolution for all cases (``RESOLUTION_LANDSAT``)?
- In principle, there is a gdal warp command used by FORCE. Could it be, that you do not have sufficient RAM, such that this operation uses the swap memory?

Cheers,
David

**Reply by maawoo** [reply]
Hi David! 
Good to hear, as that's what I thought as well. Just started processing my pool with the 10m DEM 👍🏻 

- Yes, I used the exact same parameters for processing, which are mostly the default setttings (e.g. 30m Landsat resolution)
- I actually noticed this problem on our departments HPC with ~500 GB RAM, before doing a smaller test on my Laptop... so that's also not the problem 😬

**Comment by davidfrantz** [comment]
Good choice!

I will keep this in mind though.
GDAL has some internal memory limits that are used by default - it may be that GDAL splits this operation into many small chunks.. but this is just a wild guess.

Cheers,
David
